topology/dynamic.go

```diff
@@ -113,66 +143,28 @@ func (t *dynamicTopology) Close() {
 
 	t.closed = true
 
-	t.w.Close()
-	t.s.Close()
+	t.watch.Close()
+	t.watchable.Close()
```

Comment:
Closing the watchable should also close all the watches associated with the watchable, so don't think you need to call `watch.Close()` here.